apps/frontend/src/components/searchBar.tsx

```diff
@@ -0,0 +1,252 @@
+  const isSmallPage = useMediaQuery("(min-width:740px)");
+
+  return (
+    <Grid
+      container
+      direction="row"
+      alignItems="center"
+      justifyContent="center"
+      spacing={2}
+      sx={{
+        backgroundColor: "#f5f5f5",
+        boxShadow: "0px 4px 6px #808080",
+        paddingBottom: "10px",
+      }}
+    >
+      <Grid item>
+        {isSmallPage ? (
+          <Button
+            variant="contained"
+            sx={classes.homepageButton}
+            onClick={handleReload}
+          >
+            <img
+              src={martletPlaceLogo}
+              alt="MartletPlace Logo"
+              style={{ width: "60px", marginBottom: "9px" }}
+            />
+            MartletPlace
+          </Button>
+        ) : (
+          <></>
+        )}
+      </Grid>
+      <Grid item xs={6}>
+        <TextField
+          id="outlined-basic"
+          placeholder="Search"
+          variant="outlined"
+          fullWidth
+          sx={classes.searchBar}
+          value={searchInput}
+          onChange={handleSearchInputChange}
+          onKeyDown={handleKeyDown}
+        />
+      </Grid>
+      <Grid item>
+        <Button
+          type="submit"
+          variant="contained"
+          sx={{
+            ...classes.button,
+            ...classes.whiteButton,
+            backgroundColor: showFilters ? "#808080" : "white",
+          }}
+          onClick={toggleFilters}
+        >
+          <img src={filter} alt="Filter Icon" style={{ width: "45%" }} />
+        </Button>
+      </Grid>
+      {isDesktop ? (
+        <>
+          <Grid item xs={1}>
+            <Button
+              type="submit"
+              variant="contained"
+              sx={{ ...classes.button, ...classes.yellowButton }}
+              onClick={handleSearch}
+            >
+              Search
+            </Button>
+          </Grid>
+          <Grid item xs={0.3} />
+          <Grid item>
+            <Button
+              type="submit"
+              variant="contained"
+              sx={classes.button}
+              onClick={handleMessageRoute}
+            >
+              <img src={message} alt="Message Icon" style={{ width: "45%" }} />
+            </Button>
+          </Grid>
+          <Grid item xs={1}>
+            <Button
+              type="submit"
+              variant="contained"
+              sx={classes.button}
+              onClick={handleAccountRoute}
+            >
+              Account
+            </Button>
+          </Grid>
+        </>
+      ) : (
+        <>
+          {/* Hamburger button */}
+          <Grid item xs={1}>
+            <Button
+              onClick={handleClick}
+              size="medium"
+              sx={{
+                ...classes.button,
+                fontSize: "40px",
+                maxHeight: "50px",
+                marginTop: "8px",
+                color: "white",
+              }}
+              aria-controls={open ? "account-menu" : undefined}
```

Comment:
null over undefined

Reply:
I needed to use undefined here or else the button caused errors